### PR TITLE
[UPLOAD-612] workspaces

### PR DESCRIPTION
**Pull request by krystophv**
This is a draft PR to easier illustrate the changes _since_ [UPLOAD-428]
To address [UPLOAD-612] 

[UPLOAD-428]: https://tidepool.atlassian.net/browse/UPLOAD-428
[UPLOAD-612]: https://tidepool.atlassian.net/browse/UPLOAD-612

### CODE REVIEW

File: app/actions/async.js [hunk 1]
```diff
@@ -165,35 +166,54 @@ export function doLogin(creds, opts) {
...
-        return dispatch(
-          setPage(pages.CLINIC_USER_SELECT, actionSources.USER, {
-            metric: { eventName: metrics.CLINIC_SEARCH_DISPLAYED },
-          })
-        );
-      }
+          if (clinics.length > 1) { // more than one clinic - go to workspace switch
+            return dispatch(
+              setPage(pages.WORKSPACE_SWITCH, actionSources.USER, {
+                metric: { eventName: metrics.WORKSPACE_SWITCH_DISPLAYED}
```
Comment (clintonium-119) [inline app/actions/async.js:190]:
```suggestion
                metric: { eventName: metrics.WORKSPACE_SWITCH_DISPLAYED }
```

File: app/components/Header.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ import styles from '../../styles/components/Header.module.less';
 import logo from '../../images/Tidepool_Logo_Light x2.png';
 
 import { remote } from 'electron';
+import {selectClinic} from '../actions/sync';
```
Comment (clintonium-119) [inline app/components/Header.js:37]:
```suggestion
```
This import is not actually used.

File: test/app/actions/async.test.js [hunk 3]
```diff
@@ -3722,6 +3862,249 @@ describe('Asynchronous Actions', () => {
...
+            abc123: 'US/Mountain'
+          }
+        });
+        store.dispatch(async.goToPersonalWorkspace());
+        const actions = store.getActions();
+        expect(actions).to.deep.equal(expectedActions);
+      });
+    });
+    describe('user has no supported devices', () => {
+      test('should dispatch SELECT_CLINIC, SET_UPLOAD_TARGET_USER, SET_UPLOADS, then SET_PAGE (redirect to settings page for device selection)', () => {
```
Comment (clintonium-119) [inline test/app/actions/async.test.js:3962]:
```suggestion
      test('should dispatch SELECT_CLINIC, SET_UPLOAD_TARGET_USER, then SET_PAGE (redirect to settings page for device selection)', () => {
```

**Review by clintonium-119 — APPROVED** [review]
Approved ✔️ 
Nice work.  A couple of very minor suggestions in the comments.
And thanks for putting up the smaller PR to simplify the review.

**Comment by clintonium-119** [comment]
Disregard my review comments here.  I meant to add them to the other, non-draft review.

**Comment by krystophv** [comment]
This has served its illustrative purpose

**Pull request closed without merging**